src/main/webapp/app/exercises/programming/manage/update/update-components/information/programming-exercise-information.component.ts

```diff
@@ -0,0 +1,294 @@
+        // We need to create a new object to trigger the change detection
+        this.programmingExercise().buildConfig = { ...this.programmingExercise().buildConfig };
+    }
+
+    onSolutionRepositoryCheckoutPathChange(event: string) {
+        this.programmingExercise().buildConfig!.solutionCheckoutPath = event;
+        // We need to create a new object to trigger the change detection
+        this.programmingExercise().buildConfig = { ...this.programmingExercise().buildConfig };
+    }
+
+    private registerInputFieldsWhenChildComponentsAreReady() {
+        this.shortNameField(); // triggers effect
+        this.registerInputFields();
+    }
+
+    private fetchAndInitializeTakenTitlesAndShortNames() {
+        const courseId = this.courseId() ?? this.programmingExercise().course?.id;
+        if (courseId) {
+            this.exerciseService.getExistingExerciseDetailsInCourse(courseId, ExerciseType.PROGRAMMING).subscribe((exerciseDetails: CourseExistingExerciseDetailsType) => {
+                this.alreadyUsedExerciseNames.set(exerciseDetails.exerciseTitles ?? []);
+                this.alreadyUsedShortNames.set(exerciseDetails.shortNames ?? []);
+            });
+        }
+    }
+
+    private generateShortNameWhenInSimpleMode() {
+        const shouldNotGenerateShortName = !this.isSimpleMode() || this.programmingExerciseCreationConfig.isEdit;
+        if (shouldNotGenerateShortName) {
+            this.isShortNameFromAdvancedMode.set(this.isShortNameFieldValid());
+            return;
+        }
+        let newShortName = this.exerciseTitle() ?? this.programmingExercise().title;
+        if (this.isImport() || this.isShortNameFromAdvancedMode()) {
+            newShortName = this.programmingExercise().shortName;
+        }
+
+        if (newShortName && newShortName.length > 3) {
+            const sanitizedShortName = removeSpecialCharacters(newShortName ?? '');
+            // noinspection UnnecessaryLocalVariableJS: not inlined because the variable name improves readability
+            const uniqueShortName = this.ensureShortNameIsUnique(sanitizedShortName);
+            this.programmingExercise().shortName = uniqueShortName;
+        }
+
+        this.updateIsShortNameValid();
+    }
+
+    private defineShortNameOnEditModeChangeIfNotDefinedInAdvancedMode() {
+        if (this.isSimpleMode()) {
+            this.updateIsShortNameValid();
+            this.calculateFormValid();
+        }
+    }
+
+    private updateIsShortNameValid() {
+        this.isShortNameFieldValid.set(this.getIsShortNameFieldValid());
+    }
+
+    private getIsShortNameFieldValid() {
+        return this.shortNameField() === undefined || this.shortNameField()?.control?.status === 'VALID';
+    }
+
+    private ensureShortNameIsUnique(shortName: string): string {
+        let newShortName = shortName;
+        let counter = 1;
+        while (this.alreadyUsedShortNames().includes(newShortName)) {
```

Comment:
Did you think about using a `Set` instead of includes and `for()` instead of a while loop?

Reply:
While it should not make a big difference in this case, with presumably a maximum of 50 entries, I exchanged it to use a Set instead of a List for a better code style.

We can not get rid of the while loop here, as I think it is reasonable to try a couple of names before forcing the user to come up with an own short name on the first conflict.